lighthouse-core/gather/gather-runner.js

```diff
@@ -147,18 +147,23 @@ class GatherRunner {
     });
 
     let errorDef;
+    let props;
     if (!mainRecord) {
       errorDef = LHError.errors.NO_DOCUMENT_REQUEST;
     } else if (mainRecord.failed) {
-      errorDef = {...LHError.errors.FAILED_DOCUMENT_REQUEST};
-      errorDef.message += ` ${mainRecord.localizedFailDescription}.`;
+      errorDef = LHError.errors.FAILED_DOCUMENT_REQUEST;
+      props = {
+        errorDetails: `${mainRecord.localizedFailDescription}.`,
+      };
     } else if (mainRecord.hasErrorStatusCode()) {
-      errorDef = {...LHError.errors.ERRORED_DOCUMENT_REQUEST};
-      errorDef.message += ` Status code: ${mainRecord.statusCode}.`;
+      errorDef = LHError.errors.ERRORED_DOCUMENT_REQUEST;
+      props = {
+        statusCode: `${mainRecord.statusCode}.`,
+      };
```

Comment:
this is getting complex. Can we start doing early returns on these? Should be easier to read with those.